src/views/LotteryV2/components/BuyTicketsModal/TicketInput.tsx

```diff
@@ -0,0 +1,278 @@
+    `
+    border: 1px solid #FFB237;
+    box-shadow: 0px 0px 0px 2px #FFB237;
+  `}
+  ${({ focused }) =>
+    focused &&
+    `
+    border: 1px solid #7645D9;
+    box-shadow: 0px 0px 0px 2px #E4DAF7;
+  `}
+`
+
+const DigitInput = styled.input`
+  border: none;
+  height: 32px;
+  padding: 0 12px;
+  font-size: 16px;
+  flex: 1;
+  width: 16px;
+  text-align: center;
+  min-width: 0;
+  background-color: transparent;
+  caret-color: #7a6faa;
+
+  &::placeholder {
+    text-align: center;
+  }
+
+  &:placeholder-shown {
+    text-align: left;
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  /* TODO check other browsers */
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    /* display: none; <- Crashes Chrome on hover */
+    -webkit-appearance: none;
+    margin: 0; /* <-- Apparently some margin are still there even though it's hidden */
+  }
+
+  -moz-appearance: textfield; /* Firefox */
+`
+
+const getIdLabel = (id: number): string => {
+  if (id < 10) return `#00${id}`
+  if (id < 100) return `#0${id}`
+  return `#${id}`
+}
+
+const TicketContaier: React.FC<{
+  ticket: Ticket
+  duplicateWith: number[]
+  updateTicket: UpdateTicketAction
+  disabled: boolean
+}> = ({ ticket, duplicateWith, updateTicket, disabled }) => {
+  const [focused, setFocused] = useState(false)
+  const digit1 = useRef<HTMLInputElement>(null)
+  const digit2 = useRef<HTMLInputElement>(null)
+  const digit3 = useRef<HTMLInputElement>(null)
+  const digit4 = useRef<HTMLInputElement>(null)
+  const digit5 = useRef<HTMLInputElement>(null)
+  const digit6 = useRef<HTMLInputElement>(null)
+
+  const { t } = useTranslation()
+
+  const digitRefs = [digit1, digit2, digit3, digit4, digit5, digit6]
+
+  const onPasteHandler = (e: React.ClipboardEvent) => {
+    e.preventDefault()
+    const pasteContent = e.clipboardData.getData('Text')
+    if (pasteContent.length <= 6 && /^\d+$/.test(pasteContent)) {
+      const filler = Array(6 - pasteContent.length).fill('')
+      updateTicket(ticket.id, [...pasteContent.split(''), ...filler])
+    }
+  }
+
+  const onFocusHandler = () => {
+    setFocused(true)
+  }
+
+  const onBlurHandler = () => {
+    setFocused(false)
+  }
+
+  const onChangeHandler = (event: React.KeyboardEvent, digitId: number) => {
+    const currentKey = parseInt(event.key, 10)
+
+    if (event.key === 'e' || event.key === 'E' || event.key === '.') {
+      event.preventDefault()
+    }
+
+    // Handling numberic inputs
+    if (currentKey >= 0 && currentKey <= 9) {
+      event.preventDefault()
+      const newNumbers = [...ticket.numbers]
+      newNumbers[digitId] = `${currentKey}`
+      updateTicket(ticket.id, newNumbers)
+      const nextDigitId = digitId + 1
+      // if we're not on the last digit - auto-tab
+      const nextInput = digitRefs[nextDigitId]
+      if (nextDigitId !== 6 && nextInput.current) {
+        nextInput.current.focus()
+      }
+    }
+
+    // Backsapce key
+    if (event.key === 'Backspace') {
```

Comment:
Maybe these comments are a bit redundant since you are using `Backspace` instead of the key code.

Reply:
True. Originally I was using `event.keyCode === NN` but `keyCode` seems to become deprecated so I changed to `key` but forgot to remove comments.